applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_lagrange_dirichlet_condition.cpp

```diff
@@ -0,0 +1,458 @@
+
+void MPMParticleLagrangeDirichletCondition::InitializeSolutionStep( ProcessInfo& rCurrentProcessInfo )
+{
+    MPMParticleBaseDirichletCondition::InitializeSolutionStep( rCurrentProcessInfo );
+
+    GeometryType& r_geometry = GetGeometry();
+    const unsigned int number_of_nodes = r_geometry.PointsNumber();
+    const unsigned int dimension = r_geometry.WorkingSpaceDimension();
+
+    for ( unsigned int i = 0; i < number_of_nodes; i++ )
+    {
+        array_1d<double, 3 > & lagrange_multiplier  = r_geometry[i].FastGetSolutionStepValue(VECTOR_LAGRANGE_MULTIPLIER);
+
+        for ( unsigned int j = 0; j < dimension; j++ )
+        {
+            #pragma omp atomic
+            lagrange_multiplier[j] *= 0.0;
+        }
+    }
+
+
+
+}
+
+void MPMParticleLagrangeDirichletCondition::CalculateAll(
+    MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo,
+    bool CalculateStiffnessMatrixFlag,
+    bool CalculateResidualVectorFlag
+    )
+{
+    KRATOS_TRY
+
+    const unsigned int number_of_nodes = GetGeometry().size();
+    const unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+    const unsigned int matrix_size = number_of_nodes * dimension * 2;
+
+    // Resizing as needed the LHS
+    if ( CalculateStiffnessMatrixFlag == true )
+    {
+        if ( rLeftHandSideMatrix.size1() != matrix_size )
+        {
+            rLeftHandSideMatrix.resize( matrix_size, matrix_size, false );
+        }
+
+        noalias( rLeftHandSideMatrix ) = ZeroMatrix(matrix_size,matrix_size); //resetting LHS
+    }
+
+    // Resizing as needed the RHS
+    if ( CalculateResidualVectorFlag == true ) //calculation of the matrix is required
+    {
+        if ( rRightHandSideVector.size( ) != matrix_size )
+        {
+            rRightHandSideVector.resize( matrix_size, false );
+        }
+
+        noalias( rRightHandSideVector ) = ZeroVector( matrix_size ); //resetting RHS
+    }
+
+    // Get imposed displacement and normal vector
+    const array_1d<double, 3 > & xg_c = this->GetValue(MPC_COORD);
+    const array_1d<double, 3 > & imposed_displacement = this->GetValue (MPC_IMPOSED_DISPLACEMENT);
+
+    // Prepare variables
+    GeneralVariables Variables;
+    const double augmentation_factor = this->GetValue(SCALAR_LAGRANGE_MULTIPLIER);
+    bool stabilization = this->GetValue(STABILIZATION_LAGRANGE_MULTIPLIER);
+
+    // Calculating shape function
+    Variables.N = this->MPMShapeFunctionPointValues(Variables.N, xg_c);
+    Variables.CurrentDisp = this->CalculateCurrentDisp(Variables.CurrentDisp, rCurrentProcessInfo);
+
+    // Check contact: Check contact penetration: if <0 apply constraint, otherwise no
+    bool apply_constraints = true;
+    if (Is(CONTACT))
+    {
+        // NOTE: the unit_normal_vector is assumed always pointing outside the boundary
+        array_1d<double, 3 > & unit_normal_vector = this->GetValue(MPC_NORMAL);
```

Comment:
r_ is missing

Reply:
I'll do the renaming for my new variables, but would prefer to rename those variables, which are already in the application in an additional PR for the hole application.